bayesian/utils.py

```diff
@@ -925,3 +926,80 @@ def generate_content_hash(content):
     """Return the sha1 digest of a string."""
     hash_object = hashlib.sha1(content.encode('utf-8'))
     return hash_object.hexdigest()
+
+
+def create_directory_structure(root=os.getcwd(), struct=dict()):
+    """Create a directory structure.
+
+    root: String path to root directory
+    struct: Dict object describing dir structure
+        example:
+            {
+                'name': 'parentdir',
+                'type': 'dir',
+                'contains': [
+                    {
+                        'name': 'hello.txt',
+                        'type': 'file',
+                        'contains': "Some text"
+                    },
+                    {
+                        'name': 'childdir',
+                        'type': 'dir',
+                    }
+                ]
+            }
+    """
+    _root = os.path.abspath(root)
+    if isinstance(struct, list):
+        for item in struct:
+            create_directory_structure(_root, item)
+    else:
+        # default type is file if not defined
+        _type = struct.get('type', 'file')
+        _name = struct.get('name')
+        _contains = struct.get('contains', '')
+        if _name:
+            _root = os.path.join(_root, _name)
+            if _type == 'file':
+                with open(_root, 'wb') as _file:
+                    if not isinstance(_contains, (bytes, bytearray)):
+                        _contains = _contains.encode()
+                    _file.write(_contains)
+            else:
+                os.makedirs(_root, exist_ok=True)
+                if isinstance(_contains, (list, dict)):
+                    create_directory_structure(_root, _contains)
+
+
+def push_repo(token, local_repo, remote_repo, user=None, organization=None, auto_remove=False):
+    """Initialize a git repo and push the code to the target repo."""
+    commit_msg = 'Initial commit'
+    if not os.path.exists(local_repo):
+        raise ValueError("Directory {} does not exist.".format(local_repo))
+    repo = Repo.init(local_repo)
+    repo.git.add(all=True)
+    committer = Actor(os.getenv("GIT_COMMIT_AUTHOR_NAME", "openshiftio-launchpad"),
+                      os.getenv("GIT_COMMIT_AUTHOR_EMAIL", "obsidian-leadership@redhat.com"))
+
```

Comment:
I'd suggest to refactor this function - to be universal for all commiters. And then call it with author-name + author email read from env. var. It would be more 'functional' etc.

Like:

```
def push_repo(token, local_repo, remote_repo, author_name, author_email, user=None, organization=None, auto_remove=False):
   ...
   ...
   ...
```

```
def push_repo_from_osio(token, local_repo, remote_repo, author_name, organization=None, auto_remove=False):
    author_name=os.getenv("GIT_COMMIT_AUTHOR_NAME", "openshiftio-launchpad")
    author_email=os.getenv("GIT_COMMIT_AUTHOR_EMAIL", "obsidian-
    push_repo(...)
```